### PR TITLE
Create beacon config file

### PR DESCRIPTION
**Pull request by jbydeley**
PR for issue #3 
* Add .beaconrc file
* Add loadConfig to handle loading and parsing
the .beaconrc json file

Work in Progress -- Looking for Feedback

I've included the .beaconrc but I'm wondering if that should be added to the .gitignore file. Or possibly we should be trying to load it from the users home directory? We probably should `Fatal` when we can't load the file but I don't have the prompt changes in this branch.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -90,3 +106,8 @@ func main() {
 	// Print the beacon log!
 	printLog(beaconLogData)
 }
+
+type beaconConfig struct {
```
Comment (teesloane) [inline main.go]: Would you mind moving this type to the `types` section towards the top of the file?

File: main.go [hunk 2]
```diff
@@ -71,10 +72,25 @@ func printLog(data BeaconLog) {
...
+}
+
 // ============================
 // MAIN!
 // ============================
 func main() {
+	var config beaconConfig
+
+	if err := loadConfig(&config); err != nil {
+		log.Fatal(err)
```
Comment (teesloane) [inline main.go:97]: I don't know what this does offhand (although I can guess), but I'm looking forward to learning more. I suspect we will need to implement this on the other file loading we're doing...

**Review by teesloane — COMMENTED** [review]
Looks great! just one small thing if you don't mind!

**Comment by teesloane** [comment]
Looks great! I think we should, for now, just try and load the `beaconrc` file from the main directory.

Happy to see the json parsing worked out. I think for now we can check-in the `beaconrc` as it will probably be useful to all have the same file for testing / dev'ing purposes. 

I think this should be good to merge before we get around to merging the prompt / write to file branch.

File: main.go (hunk 2 above)
Comment (jbydeley) [reply main.go:97]: They are very similar. `panic` will print the stack trace. `log.Fatal` will prepend the datetime and call `os.Exit(1)` after printing the message but will not print the stack trace.

#### log.Fatal

```
2017/08/09 23:04:17 open ./.beacosnrc: no such file or directory
exit status 1
```

#### panic
```
panic: possibly missing .beaconrc file

goroutine 1 [running]:
main.loadConfig(0xc42000a400, 0xc42000a400, 0xc42001c0a0)
        /go/src/github.com/jbydeley/beacon/main.go:79 +0x111 main.main()
        /go/src/github.com/jbydeley/beacon/main.go:92 +0x5e
exit status 2
```

Comment (teesloane) [reply main.go:97]: Ah, I like the `log.Fatal` output. Very nice, and thank you for going over that!

**Comment by teesloane** [comment]
Looks great! shipping!

**Pull request merged**